### ISSUE TITLE
Nan appears during sd3 training

### ISSUE THREAD

**Issue by jyy-1998** [huggingface/diffusers#8599]
### Describe the bug

I try to use the  train_dreambooth_sd3.py for training. The loss and results were normal at first, but after two or three thousand steps, Nan appeared. Has anyone had the same problem? I'm training with fp16 precision, but setting the accuracy of VAE to fp32 as per the training file

### Reproduction

accelerate launch --config_file config.yaml --multi_gpu train_text_to_imagesd3.py \
  --resolution=1024 \
  --train_batch_size=1 \
  --output_dir="output" \
  --num_train_epochs=10 \
  --checkpointing_steps=10000 \
  --validation_steps=500 \
  --max_train_steps=30000 \
  --learning_rate=2e-05 \
  --seed=12345 \
  --mixed_precision="fp16" \
  --use_8bit_adam \
  --gradient_accumulation_steps=1 \
  --gradient_checkpointing 

### System Info

none

**Comment by Luciennnnnnn** [comment]
I also encounter NAN problem when adapting SD3 to another task

**Comment by bghira** [comment]
please try `--max_grad_norm=1`

**Comment by bghira** [comment]
also #8592 might help

**Comment by Luciennnnnnn** [comment]
@bghira I do use these strategies, however, nan still exists.

**Comment by bghira** [comment]
can you try adding QK Norm layers back?

https://github.com/bghira/SimpleTuner/pull/469

**Comment by bghira** [comment]
this will extend the model to 3B parameters for higher performance (the 2B seems limited) and reintroduces the missing qk norm blocks

prepare for a longer training session

**Comment by Luciennnnnnn** [comment]
Hi @bghira, thank you for your suggestion! It seems be hard to train for many users by introducing extra 1B parameters, i.e. extensive computing power and longer training iterations.

**Comment by Luciennnnnnn** [comment]
I mean in order to fine-tuning a model, it is not wise to introduce at least 1B parameters. That should not be the correct direction for SD3.

**Comment by bghira** [comment]
someone has to!

**Comment by sayakpaul** [comment]
My question is what happens when you try a lower learning rate or try to overfit a single batch of training data? In any case, this seems more like a discussion to me and not an "issue".

**Comment by ultra-editing** [comment]
I also encounter NAN problem when tuning the SD3 for image editing. Is there any new updates?

**Comment by HaozheZhao** [comment]
I also encounter NAN problem when tuning the SD3 for image editing. Is there any new updates?

**Comment by sayakpaul** [comment]
Did you try the "logit_normal" weighting scheme?

**Comment by rardz** [comment]
try `--mixed_precision="bf16"` may help

**Comment by BJQ123456** [comment]
Where is this file located? train_text_to_imagesd3.py, I couldn't find it.Thanks

**Comment by sayakpaul** [comment]
We don't support full text to image fine-tuning. However, you can refer to which provides this support: https://github.com/bghira/SimpleTuner.

**Comment by sayakpaul** [comment]
Closing this since switching to "logit_normal" weighting scheme for loss resolves this issue in most cases. If not, please re-open and I will turn it into a discussion.

**Comment by order-a-lemonade** [comment]
i still get this problem when using "logit_normal" and "bf16", nan exist in the grad of fist step training, leading the parameters of transformer to be nan  after the first "optimizer.step()" . 
![image](https://github.com/user-attachments/assets/cebaf6eb-9c0d-4941-a0b5-14a68481ec53)
![image](https://github.com/user-attachments/assets/f86db921-eda6-4f89-8fdc-2364fa3a8029)

I notice that the latent shape is [4,3,130,128](4 is batch size), where my training image size is 1024x1024. Is this normal? Personally, I think [4,3,128,128] is more reasonable.

**Comment by order-a-lemonade** [comment]
> i still get this problem when using "logit_normal" and "bf16", nan exist in the grad of fist step training, leading the parameters of transformer to be nan after the first "optimizer.step()" . ![image](https://private-user-images.githubusercontent.com/66902842/378740124-cebaf6eb-9c0d-4941-a0b5-14a68481ec53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1ODcyOTMsIm5iZiI6MTcyOTU4Njk5MywicGF0aCI6Ii82NjkwMjg0Mi8zNzg3NDAxMjQtY2ViYWY2ZWItOWMwZC00OTQxLWEwYjUtMTRhNjg0ODFlYzUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA4NDk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNDhlMmI5NTBiMTllMDAxOWUyY2NkNDM1NzIxNThmZGI1NmFmNTFmYmRlYjk0MmY0ZWIzOWRkNzdiZWIwNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Hvag6l5zn9wzayNd-RCtvRISnjzzKRurhcNK7ciRrrk) ![image](https://private-user-images.githubusercontent.com/66902842/378740249-f86db921-eda6-4f89-8fdc-2364fa3a8029.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1ODcyOTMsIm5iZiI6MTcyOTU4Njk5MywicGF0aCI6Ii82NjkwMjg0Mi8zNzg3NDAyNDktZjg2ZGI5MjEtZWRhNi00Zjg5LThmZGMtMjM2NGZhM2E4MDI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA4NDk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYzg3OTIwODVlODQ4YWRjNDI5NDZiNGM3NmNlYWRhNDNmYTc1MTg4ZWUzYTlhNDgxZTQ3MzgxZWRjMDQ2Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B41vgJzU1t7Se0wft-z8cN_1cJ8ysn60Rpbyt3uGel0)

i solve this problem by changing torch version from 2.5.0 to 2.4.1. 🤣

**Comment by wangyanhui666** [comment]
> > i still get this problem when using "logit_normal" and "bf16", nan exist in the grad of fist step training, leading the parameters of transformer to be nan after the first "optimizer.step()" . ![image](https://private-user-images.githubusercontent.com/66902842/378740124-cebaf6eb-9c0d-4941-a0b5-14a68481ec53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1ODcyOTMsIm5iZiI6MTcyOTU4Njk5MywicGF0aCI6Ii82NjkwMjg0Mi8zNzg3NDAxMjQtY2ViYWY2ZWItOWMwZC00OTQxLWEwYjUtMTRhNjg0ODFlYzUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA4NDk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNDhlMmI5NTBiMTllMDAxOWUyY2NkNDM1NzIxNThmZGI1NmFmNTFmYmRlYjk0MmY0ZWIzOWRkNzdiZWIwNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Hvag6l5zn9wzayNd-RCtvRISnjzzKRurhcNK7ciRrrk) ![image](https://private-user-images.githubusercontent.com/66902842/378740249-f86db921-eda6-4f89-8fdc-2364fa3a8029.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1ODcyOTMsIm5iZiI6MTcyOTU4Njk5MywicGF0aCI6Ii82NjkwMjg0Mi8zNzg3NDAyNDktZjg2ZGI5MjEtZWRhNi00Zjg5LThmZGMtMjM2NGZhM2E4MDI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA4NDk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYzg3OTIwODVlODQ4YWRjNDI5NDZiNGM3NmNlYWRhNDNmYTc1MTg4ZWUzYTlhNDgxZTQ3MzgxZWRjMDQ2Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B41vgJzU1t7Se0wft-z8cN_1cJ8ysn60Rpbyt3uGel0)
> 
> i solve this problem by changing torch version from 2.5.0 to 2.4.1. 🤣

so is bug of pytorch 2.5.0?

**Comment by order-a-lemonade** [comment]
> > > i still get this problem when using "logit_normal" and "bf16", nan exist in the grad of fist step training, leading the parameters of transformer to be nan after the first "optimizer.step()" . ![image](https://private-user-images.githubusercontent.com/66902842/378740124-cebaf6eb-9c0d-4941-a0b5-14a68481ec53.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1ODcyOTMsIm5iZiI6MTcyOTU4Njk5MywicGF0aCI6Ii82NjkwMjg0Mi8zNzg3NDAxMjQtY2ViYWY2ZWItOWMwZC00OTQxLWEwYjUtMTRhNjg0ODFlYzUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA4NDk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkxNDhlMmI5NTBiMTllMDAxOWUyY2NkNDM1NzIxNThmZGI1NmFmNTFmYmRlYjk0MmY0ZWIzOWRkNzdiZWIwNjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Hvag6l5zn9wzayNd-RCtvRISnjzzKRurhcNK7ciRrrk) ![image](https://private-user-images.githubusercontent.com/66902842/378740249-f86db921-eda6-4f89-8fdc-2364fa3a8029.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1ODcyOTMsIm5iZiI6MTcyOTU4Njk5MywicGF0aCI6Ii82NjkwMjg0Mi8zNzg3NDAyNDktZjg2ZGI5MjEtZWRhNi00Zjg5LThmZGMtMjM2NGZhM2E4MDI5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDA4NDk1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyYzg3OTIwODVlODQ4YWRjNDI5NDZiNGM3NmNlYWRhNDNmYTc1MTg4ZWUzYTlhNDgxZTQ3MzgxZWRjMDQ2Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B41vgJzU1t7Se0wft-z8cN_1cJ8ysn60Rpbyt3uGel0)
> > 
> > 
> > i solve this problem by changing torch version from 2.5.0 to 2.4.1. 🤣
> 
> so is bug of pytorch 2.5.0?

i guess so, and the version of xformers is also important. maybe the old xformers version has conflict with the new torch version.
